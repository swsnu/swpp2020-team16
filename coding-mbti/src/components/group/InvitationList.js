/* eslint-disable react/jsx-indent */
import React from 'react';
import PropTypes from 'prop-types';

import {
  Paper, Table, TableHead, TableBody, TableCell
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import TableContainer from '@material-ui/core/TableContainer';
import TableRow from '@material-ui/core/TableRow';

import InvitationAccept from './InvitationAccept';
import InvitationDelete from './InvitationDelete';

const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
});

export default function InvitationList(props) {
  const { deleteInvitation, acceptInvitation } = props;
  const classes = useStyles();

  let { invitation } = props;
  const isInvitationDemo = Object.values(props.invitation).length === 0;
  if (isInvitationDemo) {
    const demoInvitation = {
      1: {
        id: 1, group: 'demo group SWPP', sender: 'demo sender team16'
      },
      2: {
        id: 2, group: 'demo group DC', sender: 'demo sender batman'
      },
      3: {
        id: 3, group: 'demo group marvel', sender: 'demo sender ironman'
      },
      4: {
        id: 4, group: 'demo group Amazon', sender: 'demo sender Jeff Bezos'
      },
    };
    invitation = demoInvitation;
  }

  async function handleAccept(id) {
    if (isInvitationDemo) {
      window.alert('demo invitation accept.');
      return;
    }
    await acceptInvitation(id);
  }

  async function handleDelete(id) {
    if (isInvitationDemo) {
      window.alert('demo invitation delete.');
      return;
    }
    await deleteInvitation(id);
  }

  return (
    <div>
      <Grid container spacing={6} className={classes.total}>
        <Grid item xs={12}>
          <Typography
            component="h1"
            variant="h1"
            align="center"
            color="textPrimary"
            gutterBottom
          >
            My Invitations
          </Typography>
        </Grid>
      </Grid>
      <TableContainer component={Paper}>
        <Table className={classes.table} aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>Sender</TableCell>
              <TableCell>Group</TableCell>
              <TableCell>delete</TableCell>
              <TableCell>accept</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {Object.values(invitation).map(({ id, sender, group }) => (
              <TableRow key={id}>
                <TableCell component="th" scope="row">{sender}</TableCell>
                <TableCell>{group}</TableCell>
                <TableCell>
                  <InvitationDelete
                    invitaionId={id}
                    deleteInvitation={() => handleDelete(id)}
                  />
                </TableCell>
                <TableCell>
                  <InvitationAccept
                    invitationId={id}
                    acceptInvitation={() => handleAccept(id)}
                  />
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
}

InvitationList.propTypes = {
  invitation: PropTypes.object,
  deleteInvitation: PropTypes.func.isRequired,
  acceptInvitation: PropTypes.func.isRequired,
};

InvitationList.defaultProps = {
  invitation: {}
};
