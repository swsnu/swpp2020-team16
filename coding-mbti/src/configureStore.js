import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  REGISTER,
  PAUSE,
  PERSIST,
  PURGE,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import reducer from './feature';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducer);

export default () => {
  const store = configureStore({
    reducer: persistedReducer,
    middleware: getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
  });
  const persistor = persistStore(store);
  return { store, persistor };
};
