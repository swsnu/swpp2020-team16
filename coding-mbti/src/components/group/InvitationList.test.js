import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { ThemeProvider, createMuiTheme } from '@material-ui/core';
import { transitions, positions, Provider as AlertProvider } from 'react-alert';
import AlertTemplate from 'react-alert-template-basic';
import configureStore from '../../configureStore';
import InvitationList from './InvitationList';

const theme = createMuiTheme();

const { store } = configureStore();
const options = {
    // you can also just use 'bottom center'
    position: positions.BOTTOM_CENTER,
    timeout: 5000,
    offset: '30px',
    // you can also just use 'scale'
    transition: transitions.SCALE,
};
describe('InvitationList', () => {
    let wrapper;
    const createGroup = jest.fn();

    it('render', () => {
        wrapper = mount(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <AlertProvider template={AlertTemplate} {...options}>
                        <InvitationList invitation={[]} isManager gruopId={1} createGroup={createGroup} />
                    </AlertProvider>
                </ThemeProvider>
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
