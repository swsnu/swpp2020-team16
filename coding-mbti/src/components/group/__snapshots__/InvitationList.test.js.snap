// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitationList render 1`] = `ReactWrapper {}`;
